python/eos/plot/plotter.py

```diff
@@ -63,6 +64,34 @@ def next_color(self):
         self.__next_color = (self.__next_color + 1) % len(self.colors)
         return(result)
 
+    @staticmethod
+    def weighted_quantiles(values, quantiles, sample_weight=None):
+        """ Compute the quantiles of a weighted sample, ignoring NaN values.
+        :param values: (*tuple* of *float*) -- Sample from which the quantiles are computed.
+        :param quantiles: (*tuple* of *float*) -- Quantiles to compute, the values must be in [0, 1]
+        :param sample_weight: array-like of the same length as values
+        :return: numpy.array with computed quantiles.
+        """
+        values = np.array(values)[~np.isnan(values)]
+        quantiles = np.array(quantiles)
+        if sample_weight is None:
+            sample_weight = np.ones(len(values))
+        sample_weight = np.array(sample_weight)[~np.isnan(values)]
+        if (np.any(quantiles < 0) or np.any(quantiles > 1)):
+            eos.error('Quantiles should be in [0, 1]')
+        if (np.any(sample_weight < 0)):
+            eos.error('The sample weights cannot be negative')
+        if (np.sum(sample_weight) == 0):
+            eos.error('The sum of the sample weights evaluated to zero')
+
+        sorter = np.argsort(values)
+        values = values[sorter]
+        sample_weight = sample_weight[sorter]
+
+        weighted_quantiles = np.cumsum(sample_weight) - 0.5 * sample_weight
```

Comment:
Comment would be nice to indicate that each sample's weight is half assigned to the the left and half assigned to the right of the point.